python/pdo/submitter/ccf/ccf_submitter.py

```diff
@@ -0,0 +1,589 @@
+            temp = dict()
+            temp['contract_id'] = dependency['contract_id']
+            temp['state_hash'] = crypto.base64_to_byte_array(dependency['state_hash'])
+            temp['state_hash_for_sign'] = dependency['state_hash']
+            dependencies.append(temp)
+
+        tx_params = PayloadBuilder.build_update_contract_state_transaction_from_data(
+            verb,
+            contract_enclave_id,
+            crypto.base64_to_byte_array(enclave_signature),
+            contract_id,
+            current_state_hash,
+            previous_state_hash,
+            dependencies,
+            message_hash,
+            contract_code_hash,
+            nonce = channel_keys
+            )
+
+        try:
+            response = self.submit_rpc_to_ccf(tx_method, tx_params)
+            if response.result: # result will be True for successful init transaction
+                return tx_params['nonce'] #this will represent the transaction id
+            else:
+                raise Exception(response.error['message'])
+        except Exception as e:
+            raise
+
+# -----------------------------------------------------------------
+    def get_enclave_info(self,
+        enclave_id):
+
+        tx_method = "verify_enclave_registration"
+        tx_params = PayloadBuilder.build_verify_enclave_from_data(enclave_id)
+
+        enclave_info = self.submit_read_request_to_ccf(tx_method, tx_params)
+
+        # verify ccf signature
+        message = enclave_info["verifying_key"]
+        message+= enclave_info["encryption_key"]
+        message+= enclave_info["proof_data"]
+        message+= enclave_info["last_registration_block_context"]
+        message+= enclave_info["owner_id"]
+
+        if not CCFSubmitter.verify_ledger_signature(message, enclave_info["signature"]):
+            raise Exception("Invalid signature on Get Enclave Info from CCF Ledger")
+
+        return enclave_info
+
+# -----------------------------------------------------------------
+    def get_contract_info(self,
+        contract_id):
+
+        tx_method = "verify_contract_registration"
+        tx_params = PayloadBuilder.build_verify_contract_registration_from_data(contract_id)
+
+        contract_info = self.submit_read_request_to_ccf(tx_method, tx_params)
+
+        # verify ccf signature
+        message = contract_info["contract_id"]
+        message+= contract_info["contract_code_hash"]
+        message+= contract_info["pdo_contract_creator_pem_key"]
+        for pservice_id in contract_info["provisioning_service_ids"]:
+            message+= pservice_id
+        message+= contract_info["enclaves_info"]
+
+        if not CCFSubmitter.verify_ledger_signature(message, contract_info["signature"]):
+            raise Exception("Invalid signature on Get Contract Info from CCF Ledger")
+
+        return contract_info
+
+# -----------------------------------------------------------------
+    def get_current_state_hash(self,
```

Comment:
name is misleading. You are returning the `state_info`, no the hash.

Reply:
I think the name is actually fine here, since details about the state are returned in the next `get_state_details` function. Maybe what's should be renamed is the `state_info` variable to `state_hash`.

Reply:
state_info and not state_hash since the dict also has the is_active variable in it.